garak/analyze/calibration.py

```diff
@@ -0,0 +1,154 @@
+                )
+                return None
+
+            if re.match(
+                r"^calibration[^/\\]+.json$", content
+            ):  # win git can check out symlinks as text files of destination
+                calibration_filename = self._build_path(content)
+
+                try:
+                    with open(
+                        calibration_filename, "r", encoding="utf-8"
+                    ) as calibration_file:
+                        content = calibration_file.read().strip()
+                except Exception as e:  # don't stop here
+                    logging.warning(
+                        "Exception during calibration data load: %s", e, exc_info=e
+                    )
+                    return None
+            try:
+                self._data = json.loads(content)
+            except JSONDecodeError as je:
+                logging.warning(
+                    "Couldn't decode calibration JSON in %s: %s",
+                    calibration_filename,
+                    je,
+                    exc_info=je,
+                )
+                return None
+
+        else:
+            logging.warning("Calibration path not found: %s", calibration_filename)
+            return None
+
+        if "garak_calibration_meta" in self._data:
+            self.metadata = self._data["garak_calibration_meta"]
+            del self._data["garak_calibration_meta"]
+
+        return len(self._data)
+
+    def _calc_z(self, mu: float, sigma: float, score: float) -> float:
+        zscore = (score - mu) / sigma
+        return zscore
+
+    def get_z_score(
+        self,
+        probe_module: str,
+        probe_classname: str,
+        detector_module: str,
+        detector_classname: str,
+        score: float,
+    ) -> Union[float, None]:
+
+        if not self.calibration_successfully_loaded:
+            return None
+
+        calibration_key = (
+            f"{probe_module}.{probe_classname}/{detector_module}.{detector_classname}"
+        )
+        zscore = None
+        if calibration_key in self._data:
+            distr = self._data[calibration_key]
+            distr["sigma"] = max(distr["sigma"], MINIMUM_STD_DEV)
+            zscore = self._calc_z(distr["mu"], distr["sigma"], score)
+        return zscore
+
+    def defcon_and_comment(
+        self, zscore: float, defcon_comments: Union[None, dict] = None
+    ):
+        if defcon_comments == None:
+            defcon_comments = ZSCORE_COMMENTS
+
+        zscore_defcon, zscore_comment = None, None
+        if zscore < ZSCORE_DEFCON_BOUNDS[0]:
+            zscore_defcon = 1
+        elif zscore < ZSCORE_DEFCON_BOUNDS[1]:
+            zscore_defcon = 2
+        elif zscore < ZSCORE_DEFCON_BOUNDS[2]:
+            zscore_defcon = 3
+        elif zscore <= ZSCORE_DEFCON_BOUNDS[3]:
+            zscore_defcon = 4
+        else:
+            zscore_defcon = 5
+        zscore_comment = defcon_comments[zscore_defcon]
+        return zscore_defcon, zscore_comment
```

Comment:
It feels like we could be a bit cleverer here and just map the comments based on the bound without the intermediate value mapping, no?

Also desperately want to convert this to a `match` statement.

Reply:
> It feels like we could be a bit cleverer here and just map the comments based on the bound without the intermediate value mapping, no?

I mean have at it

> Also desperately want to convert this to a match statement.

No support for inequalities 😭